storage/src/test/java/tech/pegasys/teku/storage/store/StoreTest.java

```diff
@@ -298,6 +298,40 @@ public void retrieveFinalizedCheckpointAndState() {
     assertThatThrownBy(result::get).hasCauseInstanceOf(InvalidCheckpointException.class);
   }
 
+  @Test
+  public void shouldKeepOnlyMostRecentBlocksInBlockCache() {
+    final UpdatableStore store = createGenesisStore();
+    final UInt64 epoch = UInt64.valueOf(5);
+    final UInt64 startSlot = spec.computeStartSlotAtEpoch(epoch);
+    chainBuilder.generateBlocksUpToSlot(startSlot);
+
+    // Add blocks
+    final StoreTransaction tx = store.startTransaction(new StubStorageUpdateChannel());
+    chainBuilder
+        .streamBlocksAndStates()
+        .forEach(
+            blockAndState ->
+                tx.putBlockAndState(
+                    blockAndState, spec.calculateBlockCheckpoints(blockAndState.getState())));
+    tx.commit().join();
```

Comment:
iirc we needed to do safeJoin in tests... @tbenr was looking at something around joins recently...

Reply:
Yes, we should always use `safeJoin` 'cose in some cases we may not catch issues (may not apply in this specific case but we should have the good habit)

Reply:
Got it, fixed